samples/bluetooth/hci_ipc/src/main.c

```diff
@@ -271,6 +271,14 @@ static void hci_ipc_send(struct net_buf *buf, bool is_fatal_err)
 			if (is_fatal_err) {
 				LOG_ERR("IPC service send error: %d", ret);
 			} else {
+				/* In the POSIX ARCH, code takes zero simulated time to execute,
+				 * so busy wait loops become infinite loops, unless we
+				 * force the loop to take a bit of time.
+				 *
+				 * This delay allows the IPC consumer to execute, thus making
+				 * it possible to send more data over IPC afterwards.
+				 */
+				Z_SPIN_DELAY(500);
```

Comment:
Is there a bug report or way to reproduce the issue? What does this (delay) translate to on target?

Does the commit title rightly convey the actual issue with the sample or the IPC module?

Should it be `samples: Bluetooth: hci_ipc: Fix issue using icbmsg in bsim`

Reply:
I agree about fixing the commit title :)

About 
> What does this (delay) translate to on target?
Nothing. The macro defines to empty. In target this will just busy loop like before.

> Is there a bug report or way to reproduce the issue?
Nope. But the issue is that if you try to overfill the IPC buffer (due to say a too slow host which is stalled waiting for something else) the controller would infinite loop here. Which in simulation of a 5340 would otherwise not let the netcore sleep and would not let time pass and therefore the app core to awake. This issue is now present with the new icbmsg backend, which does not try to wait during a send if there is no space, but instead just returns right away with -ENOMEM.

Reply:
The issue need to be properly addressed, it sounds from your explanation that the CPU (busy loop) will consume full active current due to a slow Host/Application. Ideally the CPU should go into system-on-idle before looping back.

Your description is a good text for a GH issue to start with, and hence improve on reducing current consumption by avoiding busy looping.

Reply:
Discussed offline with author, there is already follow up tasks to avoid any higher current consumption when using icbmsg backend.

Reply:
But there will be no additional current on real hardware. The Z_SPIN_DELAY translates to k_busy_wait only for POSIX target. For other targets (and hardware) it does nothing.

Reply:
@dchat-nordic there is a `k_yield`, practically it will end up running CPU until `ret >= 0`, right?

Reply:
This will try and try to send a message through ipc. After every send it will yield not to strave threads on the local side. In normal circumstances the ipc queue should never be 100% full for a long time.